core/src/main/java/org/apache/iceberg/util/PropertyUtil.java

```diff
@@ -70,4 +71,16 @@ public static String propertyAsString(Map<String, String> properties,
     }
     return defaultValue;
   }
+
+  public static Properties filterAndRemovePrefix(Map<String, String> properties,
```

Comment:
@kbendick I know you suggested the refactoring, but it's unclear to me how this util method can be used by others because of the return type `Properties`. I personally think `JdbcUtil` is a better place to put this, or making it a private method inside `JdbcCatalog` is also good enough. What do you think?

Reply:
Yeah that would be fine.

I had envisioned it maybe being a bit different, e.g. returning `Map<String, String>` as we do filter some maps based on prefix key and then strip off the prefix (e.g. in `initialize` calls etc). For example, we do that with Spark quite often, both when `initialize`ing the Catalog as well as merging in the `spark.sql.catalog.$(catalogName).hadoop.*` fields.

But I can agree that `JdbcUtil` would probably be just as good of a place for this. I'd be fine with moving to `JdbcUtil` or as a private method as mentioned. 👍

Reply:
It's in `PropertyUtil` because I think this func can be used elsewhere, not just JDBC catalog.
If need to change the return type to `Map<String, String>`, just let me known.
So, @jackye1995, @kbendick what is your final decision?

Reply:
I would agree with Jack that until we actually go to reuse this code, it might be best to place it in `JdbcUtil` for now @dungdm93.

Possibly as a follow up PR we could move it to be more generic and also update the various places that have similar functionality at the same time.

But for now, I think it's probably best to keep it in `JdbcUtil` so we can get this PR in to fix this bug. 🙂